### ISSUE TITLE
Display character name along with Steam profile

### ISSUE THREAD

**Issue by sapphiremagic** [tremwil/SteamP2PInfo#28]
This functionality exists in Dark Souls III's Blue Sentinel/PvP Watchdog mods, and it is very helpful for knowing which players need to be blocked. I wonder if this could be added to this program, or if it running separately from the game's .exe means that there is no way for this program to know which character belongs to which Steam profile.

**Comment by cridus** [comment]
The game's own character names aren't sent through the Steam APIs (there's no need to), so no, it's not possible, sadly.  
What you can do, is flip the value of an hidden setting via Cheat Engine (in offline mode) that makes you see Steam names in-game instead of the character names. The only downside I've seen so far is that when you check summon signs on the ground, they have no name.  
With the CE table from The Grand Archives, go under `Hero` -> `OptionData` -> `Network` and you will find `Display Player Names`. Change the value from 0 to 1.

**Comment by sapphiremagic** [comment]
I appreciate the input, cridus.

**Comment by tremwil** [comment]
I am in the process of writing an Elden Ring specific tool like this which hooks SteamNetworking IPC calls from within the Steam process instead of using event tracing. This will allow proxying game P2P traffic and thus obtaining information like character names. However, I cannot give an ETA of the release on this, as I am currently busy with other projects.

**Comment by sapphiremagic** [comment]
> I am in the process of writing an Elden Ring specific tool like this which hooks SteamNetworking IPC calls from within the Steam process instead of using event tracing. This will allow proxying game P2P traffic and thus obtaining information like character names. However, I cannot give an ETA of the release on this, as I am currently busy with other projects.

Hello. As it turns out, there is a hidden functionality in-game that can be toggled to show a player's Steam name instead of their character name, so this feature might be not worth working on. If you decide to bring over the ping filter from your DS3ConnectionInfo project, I'd love to hear about it. Thank again.

**Comment by tremwil** [comment]
Yeah, I'm aware of that setting existing, but the other approach would make the tool much more reliable, allow for much more detailed connection information, and even anti-cheat features. For example, things like checking that a player's stats are valid or telling you their RL and upgrade level would be possible. However, I'd keep the code for a P2P proxy with these abilities closed-source, as it could be used for malicious cheating as well.
